src/tokenizer.rs

```diff
@@ -422,7 +441,11 @@ impl<'a> Tokenizer<'a> {
                         s += s2.as_str();
                         return Ok(Some(Token::Number(s, false)));
                     }
-                    Ok(Some(Token::make_word(&s, None)))
+                    if s == "#" {
```

Comment:
@alamb What do you think about this change?
I'm worried about clash with lines 610:
I didn't found a better way to solve it.

`                '#' if dialect_of!(self is SnowflakeDialect) => {
                    chars.next(); // consume the '#', starting a snowflake single-line comment
                    let comment = self.tokenize_single_line_comment(chars);
                    Ok(Some(Token::Whitespace(Whitespace::SingleLineComment {
                        prefix: "#".to_owned(),
                        comment,
                    })))
                }`

Reply:
Maybe we should check if the dialect is RedshiftSqlDialect?

Reply:
The change doesn't *feel* right to me (because it effectively skips `#`s so wouldn't it mean that things like `#SELECT` will parse as `SELECT` (not just identifiers)?